### PR TITLE
Bump mlflow, fix dependabot config and set backend_store_uri input to be optional

### PR DESCRIPTION
**Pull request by YevheniiSemendiak**

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -21,6 +13,16 @@ inputs:
...
         2. add its read-write reference to the `inputs.volumes` list
       Docs: https://mlflow.org/docs/latest/tracking.html#artifact-stores
+  backend_store_uri:
+    default: ""
+    descr: |
+      URI to the storage, which should be used to dump experiments, their metrics, registered models, etc.
+      Examples:
+        Empty string: in this case the model registry functionality will not work
+        Postgress DB: postgresql://postgres:password@job-1231-123123-123.platform-jobs:5432
+        SQLite DB: sqlite:///some/path/mlflow.db - you cannot use the platform storage if it's running in Azure, but could use disk
```
Comment (anayden) [inline action.yaml:23]: `if it's running in Azure` - while this is true, I doubt regular platform users are going to know where the platform is running. How about creating a task to add a check for this scenario in the future? 
No need to do this now (seems low-priority), but in general we should warn our users IF the platform is running in Azure, not tell them "go and check where the platform is running".

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1 @@
-mlflow[extras]==1.17.0
-
-# MLFlow's 
-psycopg2-binary==2.8.6
-pymysql==1.0.2
-mysqlclient==2.0.3
```
Comment (anayden) [inline requirements.txt:6]: What if someone uses postres or mysql in the action config? Will it work?

Comment (YevheniiSemendiak) [reply requirements.txt:6]: Good catch! Added back in 99565a1.
Also, for the future, asked in MLFlow maintainers regarding those deps, m.b. we will have some reasonable info https://github.com/mlflow/mlflow/issues/4878

File: action.yaml (hunk 1 above)
Comment (YevheniiSemendiak) [reply action.yaml:23]: added in 
8e40190

**Pull request merged**